src/vs/code/node/cliProcessMain.ts

```diff
@@ -147,7 +147,7 @@ export class Main {
 		if (installedExtensionsManifests.some(manifest => isLanguagePackExtension(manifest))) {
 			await this.updateLocalizationsCache();
 		}
-		return failed.length ? Promise.reject(localize('installation failed', "Failed Installing Extensions: {0}", failed.join(', '))) : Promise.resolve();
+		return failed.length ? (Promise.reject(localize('installation failed', "Failed Installing Extensions: {0}", failed.join(', '))), process.exit(1)) : Promise.resolve();
```

Comment:
Is not the error should be handled at the root (where the main is called)? 

https://github.com/microsoft/vscode/blob/master/src/vs/code/node/cli.ts#L394

And I see the process exits with error code 1 when there are errors. Please check if this code is not being executed and if so why.

Reply:
@sandy081 , It is coming into the error part and calling `eventuallyExit` but not sure why the exit code is not set .
In my system when  I check ( `echo $?`) it is correctly set as 1.
Can you please help me here to go forward ?